base/pkg/entry.jl

```diff
@@ -342,10 +342,18 @@ function update(branch::AbstractString)
                 LibGit2.merge!(repo)
             end
         end
-        LibGit2.fetch(repo)
-        ff_succeeded = LibGit2.merge!(repo, fastforward=true)
-        if !ff_succeeded
-            LibGit2.rebase!(repo, "origin/$branch")
+        try
+            LibGit2.fetch(repo)
+            ff_succeeded = LibGit2.merge!(repo, fastforward=true)
+            if !ff_succeeded
+                LibGit2.rebase!(repo, "origin/$branch")
+            end
+        catch err
+            if isa(err, LibGit2.Error.GitError)
+                error("METADATA cannot be updated. Error: $(err.msg) [$(err.class)].\nResolve problems manually in $(Pkg.dir("METADATA")).")
```

Comment:
are "class" and "code" one-to-one? EMERGECONFLICT is potentially useful to see too